### PR TITLE
Add LEP 'Snapshot Prune'

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn#3613

### CODE REVIEW

**Review by yasker — CHANGES_REQUESTED** [review]
It seems the whole LEP is talking about `removed snapshot` but somehow referring to it as `snapshot`.

File: enhancements/20220317-snapshot-prune.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Now users just need to remove the only/latest snapshot via UI, Longhorn would reclaim almost all space used by the snapshot, which is the spec size here. 
+Then as long as users don't create a new snapshot, the actual size of this volume is the space used by the volume head only, which is up to the spec size in total.
+
+### API Changes
+N/A
+
+## Design
+### Implementation Overview
+#### longhorn-engine:
+When the snapshot purge is triggered, replica will identify if the snapshot being removed is the latest snapshot by checking one child of it is the volume head. If YES, it will start the snapshot pruning operation:
```
Comment (yasker) [inline enhancements/20220317-snapshot-prune.md:77]: We need to make sure this is done only **after** all the other snapshots have done coalescing and deletion.

**Review by yasker — APPROVED** [review]
The content LGTM. Have some grammar errors though.

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
cc @longhorn/dev for this new introduction.

**Pull request merged**